### PR TITLE
Cookie consent popup

### PR DESCRIPTION
**Pull request by mjp**
Adds `aptible.js` which contains the cookie consent logic (and can be included on other sites).

### CODE REVIEW

File: aptiblejs/. babelrc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (krallin) [inline aptiblejs/. babelrc]: There is a space in this file name - I'm guessing that's not what you want here (i.e. I guess you want `.babelrc`, not `. babelrc`).

File: aptiblejs/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "aptible.js",
+  "version": "0.0.1",
+  "description": "aptible.com frontend library",
+  "main": "index.js",
+  "scripts": {
+    "dev": "webpack --mode development --watch",
+    "build": "webpack --mode production"
```
Comment (krallin) [inline aptiblejs/package.json]: Can you please run this as part of the build in `.travis.yml` (and either don't check in the resulting asset, or make sure it's up to date?)?

As-is, we might end up forgetting to rebuild the asset when making changes

File: aptiblejs/src/analytics.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
+export const utmVars = ['utm_source', 'utm_medium', 'utm_campaign', 'utm_content'];
+
+export function allParams() {
+  let params = {};
```
Comment (krallin) [inline aptiblejs/src/analytics.js]: (nit) if you're not going to re-assign it, make it `const`.

File: aptiblejs/src/analytics.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+export function identify(email) {
+  if (window.analytics) {
+    window.analytics.identify(email, {
+      email: email,
+      url: currentURL()
+    });
+  }
+}
+
+export function firePixel(service) {
```
Comment (krallin) [inline aptiblejs/src/analytics.js]: Is this used anywhere?

File: aptiblejs/src/analytics.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+export function event(name) {
+  if (window.analytics) {
+    let payload = allUtmVars();
+    payload['url'] = currentURL();
+
+    window.analytics.track(name, payload);
+  }
+}
+
+export function identify(email) {
```
Comment (krallin) [inline aptiblejs/src/analytics.js]: Do we use this anywhere?

File: aptiblejs/src/cookie.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  return cookies;
+}
+
+export function get(name) {
+  return all()[name];
+}
+
+export function write(name, value, expiresAt) {
+  let cookieString = `${name}=${value}; expires=${expiresAt.toUTCString()};`;
+  cookieString += `domain=.${window.location.hostname.replace('www.', '')}; path=/`;
```
Comment (krallin) [inline aptiblejs/src/cookie.js]: Not the biggest fan of assuming the domain is `www.` - can we make this a constant somewhere so it's a little more explicit? Alternatively, check the URL starts with `www.`?

**Comment by krallin** [comment]
@mjp you might want to avoid making PRs from your master branch in general, though ;)

File: aptiblejs/src/analytics.js (hunk 4 above)
Comment (mjp) [reply aptiblejs/src/analytics.js]: This is (will be) used on paid landing pages called via the public API `aptible.analytics.firePixel()`

File: aptiblejs/src/analytics.js (hunk 5 above)
Comment (mjp) [reply aptiblejs/src/analytics.js]: This is (will be) used on paid landing pages called via the public API `aptible.analytics.identify()`

**Pull request merged**